sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,645 @@
+ *         return -1;
+ *     }
+ *
+ *     while (1) {
+ *         int res;
+ *         char buf[128];
+ *         sock_dtls_session_t session;
+ *
+ *         res = sock_dtls_recv(&dtls_sock, &session, buf, sizeof(buf),
+ *                              SOCK_NO_TIMEOUT);
+ *         if (res > 0) {
+ *             printf("Received %d bytes: %*.s\n", res, res, buf);
+ *             if (sock_dtls_send(&dtls_sock, &session, buf, res) < 0) {
+ *                 puts("Error sending reply");
+ *             }
+ *         }
+ *     }
+ *     return 0;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * This is an example of a DTLS echo server.
+ *
+ * To create a DTLS sock, we first need to have an initialized UDP sock.
+ * DTLS sock will inherit the port from given UDP sock. So the port to be used
+ * later to listen to incoming DTLS packets will need to be set here already.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * sock_udp_t udp_sock;
+ * sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ * local.port = 20220;
+ * if (sock_udp_create(&udp_sock, &local, NULL, 0) < 0) {
+ *     puts("Error creating UDP sock");
+ *     return -1;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Using the initialized UDP sock, we can then create our DTLS sock. We use
+ * SOCK_DTLS_SERVER_TAG, which is defined as `10` in our application code
+ * beforehand, as our tag. Using @ref SOCK_DTLS_1_2 and @ref SOCK_DTLS_SERVER
+ * we set our DTLS endpoint to use DTLS version 1.2 and act as a DTLS server.
+ *
+ * Note that some DTLS implementation does not support earlier versions of DTLS.
+ * In this case, @ref sock_dtls_create() will return an error. A list of
+ * supported DTLS version for each DTLS implementation can be found at this
+ * [page](@ref net_dtls). In case of error, the program is stopped.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #define SOCK_DTLS_SERVER_TAG (10)
+ *
+ * [...]
+ *
+ * sock_dtls_t dtls_sock;
+ * if (sock_dtls_create(&dtls_sock, &udp_sock,
+ *                      SOCK_DTLS_SERVER_TAG,
+ *                      SOCK_DTLS_1_2, SOCK_DTLS_SERVER) < 0) {
+ *     puts("Error creating DTLS sock");
+ *     return -1;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Now we can listen to incoming packets using @ref sock_dtls_recv(). The
+ * application waits indefinitely for new packets. If we want to timeout this
+ * waiting period we could alternatively set the `timeout` parameter of the
+ * function to a value != @ref SOCK_NO_TIMEOUT. If an error occurs we just
+ * ignore it and continue looping. We can reply to an incoming message using
+ * its `session`.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * while (1) {
+ *     int res;
+ *     char buf[128];
+ *     sock_dtls_session_t session;
+ *
+ *     res = sock_dtls_recv(&dtls_sock, &session, buf, sizeof(buf),
+ *                          SOCK_NO_TIMEOUT);
+ *     if (res > 0) {
+ *         printf("Received %d bytes: %s -- echo!\n", res, buf);
```

Comment:
Even worse. This could potentially print anything (e.g. secret keys) after `buf` if `buf` is not terminated by a `\0`

Reply:
I'll limit the print to the length of received bytes. More on this in the above comment.